src/chapters/aspdotnet-model-classes/model-class.rst

```diff
@@ -0,0 +1,214 @@
+
+.. sourcecode:: guess
+   :lineno-start: 22
+
+   @foreach (var evt in ViewBag.events)
+   {
+      <tr>
+         <td>@evt.Name</td>
+      </tr>
+   }
+
+.. admonition:: Tip
+
+   Here's a shorthand to create auto-implementing properties. In a class, type the word “prop” followed 
+   by hitting the Tab key twice. This swiftly supplies the property’s scaffolding:
+
+   .. sourcecode:: c#
+
+      public object MyProperty { get; set; }
+
+
+Add a Model Property - Video
+----------------------------
+
+.. TODO: Add adding model property video
+
+.. topics covered: add description property to event model, update index view and post handler 
+.. to display the property and use it to create the object
+
+YOUTUBE VIDEO HERE
+
+.. admonition:: Note
+
+   The starter code for this video is found at the `create-model branch <https://github.com/LaunchCodeEducation/CodingEventsDemo/tree/create-model>`__
+   of ``CodingEventsDemo``. The final code presented in this 
+   video is found on the `add-property branch <https://github.com/LaunchCodeEducation/CodingEventsDemo/tree/add-property>`__.
+
+Add a Model Property - Text
+---------------------------
+
+To round out the ``Event`` class, we'll add a ``Description`` property to showcase what our events are all about.
+
+Updates to ``Models/Event.cs``:
+
+.. sourcecode:: c#
+   :linenos:
+
+   namespace CodingEventsDemo.Models
+   {
+      public class Event
+      {
+         public string Name { get; set; }
+
+         public Event(string name)
+         {
+            Name = name;
+         }
+
+         public override string ToString()
+         {
+            return Name;
+         }
+      }
+   }
+
+Now that our data is object-oriented, it’s quick and easy to add a new property affiliated with an event. 
+If we decide to add properties, such as ``Date`` or ``Location``, we can follow the pattern established. 
+Think about when we stored events as key-value pairs. At that stage, more significant changes were necessary 
+to add fields.
+
+In the ``Views`` folder, the ``Events/Add.cshtml`` template still uses a ``desc`` field so we don't need to update
+this view. We do, however, need to do into ``Events/Index.cshtml`` to add the table data for an event's description.
+
+``Events/Index.cshtml``:
+
+.. sourcecode:: guess
+   :lineno-start: 26
+
+   <td>@evt.Description</td>
+
+Lastly, add a parameter to the ``NewEvent`` action method. This parameter passes the description value into 
+the creation of the ``Event`` object.
+
+``EventController``:
+
+.. sourcecode:: c#
+   :lineno-start: 30
+
+   [HttpPost]
+   [Route("Events/Add")]
+   public IActionResult NewEvent(string name, string desc)
+   {
+      Events.Add(new Event(name, desc));
+
+      return Redirect("/Events");
+   }
+
+
+Check Your Understanding
+-------------------------
+
+.. admonition:: Question
+
+   True/False: Model code is framework independent.
```

Comment:
I would rephrase this. It took me a second to get where you were going with this. I think we talk about so much being wrapped up in the framework, that I did a bit of a double-take. Just food for thought!

Reply:
i think a double take is good!